### PR TITLE
Added ProtocolOwnedFarming and Whitelisted Cauldron

### PR DESCRIPTION
**Pull request by 0xm3rlin**

### CODE REVIEW

File: src/cauldrons/CauldronV4.sol [hunk 1]
```diff
@@ -152,12 +152,13 @@ contract CauldronV4 is BoringOwnable, IMasterContract {
         require(address(collateral) != address(0), "Cauldron: bad pair");
 
         magicInternetMoney.approve(address(bentoBox), type(uint256).max);
-        (, exchangeRate) = oracle.get(oracleData);
 
         blacklistedCallees[address(bentoBox)] = true;
         blacklistedCallees[address(this)] = true;
         blacklistedCallees[BoringOwnable(address(bentoBox)).owner()] = true;
 
+        (, exchangeRate) = oracle.get(oracleData);
```
Comment (0xCalibur) [inline src/cauldrons/CauldronV4.sol:160]: Should we rename is to CauldronV5.sol ?

File: src/cauldrons/WhitelistedCauldronV4.sol [hunk 2]
```diff
@@ -0,0 +1,45 @@
+// SPDX-License-Identifier: UNLICENSED
```
Comment (0xCalibur) [inline src/cauldrons/WhitelistedCauldronV4.sol:1]: This is the same as https://github.com/Abracadabra-money/magic-internet-money/blob/main/contracts/WhitelistedCauldronV3.sol ? But using hooks ?

Comment (0xm3rlin) [reply src/cauldrons/WhitelistedCauldronV4.sol:1]: yes exactly

Comment (0xm3rlin) [reply src/cauldrons/WhitelistedCauldronV4.sol:1]: makes it cleaner and much easier to upgrade

File: src/cauldrons/CauldronV4.sol (hunk 1 above)
Comment (0xm3rlin) [reply src/cauldrons/CauldronV4.sol:160]: i feel like that might not be worth it, considering all libraries would have to be updated.
Adding the fix seems most necessary

**Review by 0xCalibur — APPROVED** [review]

**Pull request merged**